### DISCUSSION TITLE
Direction offset in animation editor and preview on sprite

### DISCUSSION THREAD

**Discussion by da-rulez** [AscensionGameDev/Intersect-Engine#2028] [Ideas]
Right now the animation system works perfectly fine for animations that play directly on the player, but making animations for spells that need to rotate and aren't centered is often very hard, and sometimes impossible. 

I propose that we make this easier, and at the same time make troubleshooting any animation easier as well: 

New option to display an entity in the animation preview box, so you can see how it will look like on that entity (obviously this would need to render in the same way as it does ingame) -> Preview Sprite button is where you can set a graphic.

This selection works similar to setting an event graphic, you can select any frame from any entity animation: 

![image](https://github.com/AscensionGameDev/Intersect-Engine/assets/26048926/e0d305d0-5a8f-4424-aecf-061a70811e87)

Another option to set offsets depending on the direction of the player.

Green outlined is where the user can manage these offsets. The list is auto-populated with the directions a player can look in (4 or 8 directions, depending on the server settings) 

![image](https://github.com/AscensionGameDev/Intersect-Engine/assets/26048926/c898ed21-676a-42c2-b243-12ce4de3265f)

Outlined in red is the original Revert Changes button which is now made smaller. Add a tooltip that says: Revert changes

**Comment by da-rulez** [comment]
![image](https://github.com/AscensionGameDev/Intersect-Engine/assets/26048926/1507f363-e98e-41e8-92b2-d819bb7cf8f4)
Sprite Preview -> Pick which sprite you want to preview on, will work like the event editor where you can pick a specific spriteframe like in here: 
![image](https://github.com/AscensionGameDev/Intersect-Engine/assets/26048926/0c7cc96d-b1fc-4e14-bbd8-3e6d24e1ef0c)

On the upper layer in the first screenshot I have selected a standard spriteframe, that will now be rendered (slightly opaque possibly) exactly how the animation will play ingame as well. 

The direction picked is south, so the edits you make now in terms of offset, will only happen when the players direction is south as well. 

Offset is a pixel offset where the animation will play on the character. 

There is an option to have the same offset for the light effects so they won't render at the wrong place. Not sure if it needs to be optional actually.

**Comment by da-rulez** [comment]
Small changes in layout: 

Undo button -> Needs tooltip to show it reverts changes in lighting setup
I was thinking you can double click the big white preview square to select a sprite to include in the preview -> Maybe a tooltip when mouse-over the white square. 

Scrollable list view of all the 4 (or 8) directions

![image](https://github.com/AscensionGameDev/Intersect-Engine/assets/26048926/4588a940-9502-4853-a09a-37a4b8356fa9)

**Comment by misbahjabeen931** [comment]
learn graphic for online 
https://w2beginner.com/graphic-designing-course-in-rawalpindi/
